airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/core/IntegrationCommand.kt

```diff
@@ -0,0 +1,170 @@
+import io.airbyte.cdk.core.operation.OperationType
+import io.airbyte.cdk.integrations.base.AirbyteTraceMessageUtility
+import io.airbyte.cdk.integrations.base.JavaBaseConstants
+import io.airbyte.cdk.integrations.util.ApmTraceUtils
+import io.airbyte.cdk.integrations.util.ConnectorExceptionUtil
+import io.airbyte.protocol.models.v0.AirbyteConnectionStatus
+import io.airbyte.protocol.models.v0.AirbyteMessage
+import io.github.oshai.kotlinlogging.KotlinLogging
+import io.micronaut.context.annotation.Value
+import jakarta.inject.Inject
+import jakarta.inject.Named
+import java.nio.file.Path
+import java.util.Optional
+import java.util.function.Consumer
+import picocli.CommandLine
+
+private val logger = KotlinLogging.logger {}
+
+/** CLI implementation that invokes the requested operation for the connector. */
+@CommandLine.Command(
+    name = "airbyte-connector",
+    description = ["Executes an Airbyte connector"],
+    mixinStandardHelpOptions = true,
+    header =
+        [
+            "@|magenta     ___    _      __          __       |@",
+            "@|magenta    /   |  (_)____/ /_  __  __/ /____   |@",
+            "@|magenta   / /| | / / ___/ __ \\/ / / / __/ _   |@",
+            "@|magenta  / ___ |/ / /  / /_/ / /_/ / /_/  __/  |@",
+            "@|magenta /_/  |_/_/_/  /_.___/\\__, /\\__/\\___/|@",
+            "@|magenta                    /____/              |@",
+        ],
+)
+@SuppressFBWarnings(
+    value = ["NP_NONNULL_RETURN_VIOLATION", "NP_OPTIONAL_RETURN_NULL"],
+    justification = "Uses dependency injection",
+)
+class IntegrationCommand : Runnable {
+    @Value("\${micronaut.application.name}") lateinit var connectorName: String
+
+    @Inject lateinit var operations: List<Operation>
+
+    @Inject
+    @Named("outputRecordCollector")
+    lateinit var outputRecordCollector: Consumer<AirbyteMessage>
+
+    @CommandLine.Parameters(
+        index = "0",
+        description =
+            [
+                "The command to execute (check|discover|read|spec|write)",
+                "\t check - checks the config can be used to connect",
+                "\t discover - outputs a catalog describing the source's catalog",
+                "\t read - reads the source and outputs messages to STDOUT",
+                "\t spec - outputs the json configuration specification",
+                "\t write - writes messages from STDIN to the integration",
+            ],
+    )
+    lateinit var command: String
+
+    /*
+     * This option is present so that the usage information will include the option.  It
+     * is not actually used in this class and is handled by the property source loader.
+     */
+    @CommandLine.Option(
+        names = ["--" + JavaBaseConstants.ARGS_CONFIG_KEY],
+        description =
+            [
+                JavaBaseConstants.ARGS_CONFIG_DESC,
+                "Required by the following commands: check, discover, read, write",
+            ],
+    )
+    lateinit var configFile: Optional<Path>
```

Comment:
Should we just stop using optional, now that stuff is implemented in kotlin? I feel like it's redundant with the `?` that's now available to us, no?

Reply:
I guess we can't avoid the lateinit either ?

Reply:
Correct.  This is specific to this class because of how PicoCLI works.  In general, your statement is correct (avoid Optional).